modules/reactor/src/meshgenerators/CoreMeshGenerator.C

```diff
@@ -176,6 +176,16 @@ CoreMeshGenerator::CoreMeshGenerator(const InputParameters & parameters)
     // Skip if assembly name is equal to dummy assembly name
     if (_inputs[i] == _empty_key)
       continue;
+
+    // Save properties of first non-dummy assembly to compare to other assemblies
+    if (first_nondummy_assembly == "")
```

Comment:
this is equivalent to i == 0 right here?

Reply:
Not quite, this will get triggered if all input assemblies are set to be dummy assemblies, since this block won't get touched:

```
    // Save properties of first non-dummy assembly to compare to other assemblies
    if (first_nondummy_assembly == "")
    {
      first_nondummy_assembly = MeshGeneratorName(_inputs[i]);
      reactor_params =
          MeshGeneratorName(getMeshProperty<std::string>(RGMB::reactor_params_name, _inputs[i]));
      assembly_homogenization = getMeshProperty<bool>(RGMB::is_homogenized, _inputs[i]);
      pin_as_assembly = getMeshProperty<bool>(RGMB::is_single_pin, _inputs[i]);
    }
```
since the code will branch to the `continue` statement at each iteration